ext/RMagick/rmimage.c

```diff
@@ -4104,14 +4484,68 @@ Image_morphology_channel(VALUE self, VALUE channel_v, VALUE method_v, VALUE iter
 
     exception = AcquireExceptionInfo();
 
+#if defined(IMAGEMAGICK_7)
+    BEGIN_CHANNEL_MASK(image, channel);
+    new_image = MorphologyImage(image, method, NUM2LONG(iterations), kernel, exception);
+    CHANGE_RESULT_CHANNEL_MASK(new_image);
+    END_CHANNEL_MASK(image);
+#else
     new_image = MorphologyImageChannel(image, channel, method, NUM2LONG(iterations), kernel, exception);
+#endif
     rm_check_exception(exception, new_image, DestroyOnError);
     (void) DestroyExceptionInfo(exception);
 
     rm_ensure_result(new_image);
     return rm_image_new(new_image);
 }
 
+#if defined(IMAGEMAGICK_7)
+// TODO: Move this to KernelInfo class as a constructor?
```

Comment:
Need to fix TODO ?

Reply:
I added this as a TODO because I think this should be done after IM7 support has been added. I can remove the TODO if you want me to.

Reply:
Should we maybe open an issue for this instead?

Reply:
Lets open it once this has merged?